pydicom/data/download.py

```diff
@@ -51,53 +65,107 @@ def calculate_file_hash(filename):
     return hasher.hexdigest()
 
 
-def get_config_dir():
+def get_config_dir() -> str:
+    """Return the path to the pydicom config directory, creating it if required
+
+    The config directory will be named ``.pydicom`` and will be created in the
+    local user's home directory.
+    """
     config_dir = pathlib.Path.home().joinpath(".pydicom")
     config_dir.mkdir(exist_ok=True)
 
     return config_dir
 
 
 @retry.retry(urllib.error.HTTPError)
-def download_with_progress(url, filepath):
+def download_with_progress(url: str, fpath: pathlib.Path) -> None:
+    """Download the file at `url` to `fpath` with a progress bar.
+
+    Parameters
+    ----------
+    url : str
+        The URL to download the file from.
+    fpath : pathlib.Path
+        The absolute path where the file will be written to.
+    """
     with DownloadProgressBar(
         unit="B", unit_scale=True, miniters=1, desc=url.split("/")[-1]
     ) as t:
-        urllib.request.urlretrieve(url, str(filepath), reporthook=t.update_to)
+        urllib.request.urlretrieve(
+            url, os.fspath(fpath), reporthook=t.update_to
+        )
 
 
-def get_data_dir():
+def get_data_dir() -> str:
+    """Return the path to the cache directory, creating it if required."""
     data_dir = get_config_dir().joinpath("data")
     data_dir.mkdir(exist_ok=True)
 
     return data_dir
 
 
 @functools.lru_cache()
-def get_url_map():
-    with open(str(HERE.joinpath("urls.json")), "r") as url_file:
-        url_map = json.load(url_file)
+def get_url_map() -> Dict[str, str]:
+    """Return a dict containing the URL mappings from ``urls.json```."""
+    with open(os.fspath(HERE.joinpath("urls.json")), "r") as url_file:
+        return json.load(url_file)
+
 
-    return url_map
+def get_url(filename: str) -> str:
+    """Return the download URL corresponding to `filename`.
 
+    The filename:URL mappings are located in the ``urls.json`` file.
 
-def get_url(filename):
-    url_map = get_url_map()
+    Parameters
+    ----------
+    filename : str
+        The filename of the file to get the corresponding URL for.
 
+    Returns
+    -------
+    str
+        The download URL corresponding to `filename`.
+
+    Raises
+    ------
+    KeyError
```

Comment:
Rightly or wrongly, this one actually raises a ValueError

Reply:
Whoops